src/Hooks/Interacter.php

```diff
@@ -0,0 +1,221 @@
+<?php
+
+namespace Pantheon\Terminus\Hooks;
+
+use Pantheon\Terminus\Config\ConfigAwareTrait;
+use Pantheon\Terminus\Exceptions\TerminusException;
+use Robo\Contract\ConfigAwareInterface;
+use Consolidation\AnnotatedCommand\AnnotationData;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use League\Container\ContainerAwareInterface;
+use League\Container\ContainerAwareTrait;
+use Symfony\Component\Console\Input\InputArgument;
+use Pantheon\Terminus\Session\SessionAwareInterface;
+use Pantheon\Terminus\Session\SessionAwareTrait;
+
+
+/**
+ * Class Interacter
+ * @package Pantheon\Terminus\Hooks
+ */
+class Interacter implements ConfigAwareInterface, ContainerAwareInterface, SessionAwareInterface
+{
+    use ConfigAwareTrait;
+    use ContainerAwareTrait;
+    use SessionAwareTrait;
+
+    /**
+     * Gets required arguments in an interactive way.
+     * The Annotated Commands hook manager will call this function during the interact phase
+     * of any command that has an 'interact' annotation.
+     * 
+     * Possible tag values:
+     * - only-required-arguments: Only required arguments will be asked for. (default)
+     * - all-arguments: All arguments will be asked for.
+     * - all: All arguments and options will be asked for.
+     * 
+     * Also configurable through the 'interact-options-exclude' tag to exclude presenting options for the excluded arguments/options.
+     *
+     * @hook interact @interact
+     *
+     * @throws TerminusException
+     */
+    public function interact(InputInterface $input, OutputInterface $output, AnnotationData $annotationData)
+    {
+        $command_name = $annotationData->get('command');
+        if (empty($command_name)) {
+            // Nothing to do if no command name is provided.
+            return;
+        }
+
+        $interact_type = $annotationData->get('interact');
+        if (empty($interact_type)) {
+            // Default to only-required-arguments if no interact type is provided.
+            $interact_type = 'only-required-arguments';
+        }
+        if (!in_array($interact_type, ['only-required-arguments', 'all-arguments', 'all'])) {
+            throw new TerminusException('Invalid interact type: {type}', ['type' => $interact_type]);
+        }
+
+        $interact_options_exclude = $annotationData->get('interact-options-exclude');
+        if (!empty($interact_options_exclude)) {
+            $interact_options_exclude = explode(',', $interact_options_exclude);
+        }
+        if (!is_array($interact_options_exclude)) {
+            $interact_options_exclude = [];
+        }
+
+        $io = new SymfonyStyle($input, $output);
+        $container = $this->getContainer();
+        $application = $container->get('application');
+        $command = $application->find($command_name);
```

Comment:
could we get the command class and call functions from it to fill in the options?

Reply:
Doable like this, but not implementing it right now:

```
$command = $application->find($command_name);
        $names = $command->getNames();
        $command_class = '';
        if (count($names) > 1) {
            $command_class = $names[1];
            if (!class_exists($command_class)) {
                $command_class = '';
            }
        }
        $definition = $command->getNativeDefinition();
```